airflow/providers/microsoft/azure/hooks/data_factory.py

```diff
@@ -477,12 +479,112 @@ def delete_dataset(
         Delete the dataset.
 
         :param dataset_name: The dataset name.
-        :param resource_group_name: The dataset name.
+        :param resource_group_name: The resource group name.
         :param factory_name: The factory name.
         :param config: Extra parameters for the ADF client.
         """
         self.get_conn().datasets.delete(resource_group_name, factory_name, dataset_name, **config)
 
+    @provide_targeted_factory
+    def get_dataflow(
+        self,
+        dataflow_name: str,
+        resource_group_name: Optional[str] = None,
+        factory_name: Optional[str] = None,
+        **config: Any,
+    ) -> DataFlow:
+        """
+        Get the dataflow.
+
+        :param dataflow_name: The dataflow name.
+        :param resource_group_name: The resource group name.
+        :param factory_name: The factory name.
+        :param config: Extra parameters for the ADF client.
+        :return: The dataflow.
+        """
+        return self.get_conn().data_flows.get(resource_group_name, factory_name, dataflow_name, **config)
+
+    def _dataflow_exists(self, resource_group_name, factory_name, dataflow_name) -> bool:
```

Comment:
Can we type annotate the arguments here? Also, since `resource_group_name` & `factory_name` could be `None`, does the `list_by_factory` method work well? Or do we need to handle the case separately for `None` values?

Reply:
Done with the change in latest commit , resource_group_name & factory_name can be passed by the below lines if not passed.
```

        bind_argument("resource_group_name", "extra__azure_data_factory__resource_group_name")
        bind_argument("factory_name", "extra__azure_data_factory__factory_name")
```